### DISCUSSION TITLE
What is current license for FileManager?

### DISCUSSION THREAD

**Discussion by AntonKirilov** [serghey-rodin/vesta#2297] [General]
Hello everyone,
I saw that licenses for FileManager and SFTP chroot are free now.
 
![image](https://github.com/outroll/vesta/assets/65451492/404cca6c-8537-4452-85f9-1ed8ddb8dab6)

I have a question for maintainer @outrolled - what is the current license for those plugins - are they been licensed under GPLv 3.0 since 25 February 2024 ?

And since VestaCP has 1000 forks on GitHub, are they allowed to enable those plugins in their forks automatically?

**Comment by outrolled** [comment]
@AntonKirilov the plugins are free to use on any compatible fork that relies on vesta for license verification, however plugins are not covered under the GPLv3 license. We will clarify licensing soon, but my goal is for the core product to always be GPLv3, and plugins are add-ons that can be free, or paid.
I still want to consult the community, but my idea is:
- SFTP CHROOT and File Manager = Free, open source (maybe GPL if the community wants)
- CI/CD Automation = Free, open source but not GPL (maybe BSL 1.1?)
- Container Management = Paid, open source not GPL (maybe BSL 1.1?)
- Billing management = paid, closed source (maybe as a service?)

I'm flexible with the above, but my goal is to ensure that there are advantages to using Vesta that will also be an incentive to have more contributors to the core product. The paid plugins are a way to get some income that will go 100% back into Vesta.
Please let me know what you think, or if you don't agree with the above.

**Comment by myvesta** [comment]
Hi @outrolled ,
We are truly sorry that you are leaving the old code base, but we completely understand why you are doing it. If we had enough time, we too would write the entire hosting panel from scratch.
But then it would no longer be "VestaCP".

Anyway, we'd love to hear your thoughts on the following.

We may be wrong, but our assessment is that the current FileManager code will probably not be used in your new code base.
We would be sorry if that code went unused, especially since you gave everyone the license for free.

In the past, we cooperated well, the myVesta fork and @dpeca transferred a lot of patches from myVesta to the official VestaCP.

We would like to ask you if you would give us permission to activate the FileManager license on all existing installations (through Update) and all future installations.
We could write in the left sidebar or top header that FileManager was donated from the official VestaCP hosting panel.

**Reply by outrolled** [reply]
Hi @myvesta thank you for reaching out - sure, please send us an email on info@vestacp.com and we can work out the details.
